src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/LdapAccessPolicyDAO.scala

```
@@ -42,6 +45,18 @@ class LdapAccessPolicyDAO(protected val ldapConnectionPool: LDAPConnectionPool,
     ldapConnectionPool.delete(resourceDn(resource))
   }
 
+  override def loadResourceAuthDomain(resource: Resource): Future[Set[WorkbenchGroupName]] = Future {
+    ldapSearchStream(resourceDn(resource), SearchScope.BASE, Filter.createEqualityFilter("objectclass", ObjectClass.resource))(unmarshalResource).toSet[Resource].flatMap(_.authDomain)
```

use `ldapConnectionPool.getEntry` instead of `ldapSearchStream`, this should load a specific entity, not do a search

as the code stands, it will return an empty set if the resource is not found or does not have the right object class which probably hides the error